pkg/linode-bs/limits.go

```diff
@@ -0,0 +1,31 @@
+package linodebs
+
+// maxVolumeAttachments returns the maximum number of block storage volumes
+// that can be attached to a Linode instance, given the amount of memory the
+// instance has.
+//
+// TODO: This code should be cleaned up to use the built-in max and min
+// functions once the project is updated to Go 1.21. See
+// https://go.dev/ref/spec#Min_and_max.
+func maxVolumeAttachments(memoryBytes uint) int {
+	attachments := memoryBytes >> 30
+	if attachments > maxAttachments {
+		return maxAttachments
+	}
+	if attachments < maxPersistentAttachments {
+		return maxPersistentAttachments
+	}
+	return int(attachments)
+}
+
+const (
+	// maxPersistentAttachments is the default number of volume attachments
+	// allowed when they are persisted to an instance/boot config. This is
+	// also the maximum number of allowed volume attachments when the
+	// instance type has < 16GiB of RAM.
+	maxPersistentAttachments = 8
```

Comment:
previously, we had to set this to 7, because it conflicted with the root disk (root + 7 attached). Is 8 going to work?

Reply:
8 will work, only because `maxVolumeAttachments` is written to objectively return the number of volumes that _can be attached_ to a Linode, given the amount of RAM on the instance. The returned value is used to perform the subjective check in [`pkg/linode-bs.LinodeControllerServer.canAttach`](https://github.com/linode/linode-blockstorage-csi-driver/pull/167/files#diff-a45fd181b265bc173ff27928709318c12e64ee8eedb87832081ef107dbc96197R301-R309).

In other words, `maxVolumeAttachments` will say "a Linode with 1GiB of RAM can have 8 volumes attached". `LinodeControllerServer.canAttach` will get the list of volumes currently attached to a Linode, and indicate if there is room for another attachment.